core/services/relay/evm/method_binding.go

```diff
@@ -79,14 +87,44 @@ func (m *methodBinding) GetLatestValue(ctx context.Context, params, returnValue
 		Data: data,
 	}
 
-	bytes, err := m.client.CallContract(ctx, callMsg, nil)
+	block, err := m.blockNumberFromConfidence(ctx, confidenceLevel)
+	if err != nil {
+		return err
+	}
+
+	bytes, err := m.client.CallContract(ctx, callMsg, block)
 	if err != nil {
 		return fmt.Errorf("%w: %w", commontypes.ErrInternal, err)
 	}
 
-	return m.codec.Decode(ctx, bytes, returnValue, wrapItemType(m.contractName, m.method, false))
+	return m.codec.Decode(ctx, bytes, returnVal, WrapItemType(m.contractName, m.method, false))
 }
 
 func (m *methodBinding) QueryKey(_ context.Context, _ query.KeyFilter, _ query.LimitAndSort, _ any) ([]commontypes.Sequence, error) {
 	return nil, nil
 }
+
+func (m *methodBinding) blockNumberFromConfidence(ctx context.Context, confidenceLevel primitives.ConfidenceLevel) (*big.Int, error) {
+	confirmations, err := confidenceToConfirmations(m.confirmationsMapping, confidenceLevel)
+	if err != nil {
+		err = fmt.Errorf("%w for contract: %s, method: %s", err, m.contractName, m.method)
+		if confidenceLevel == primitives.Unconfirmed {
+			m.lggr.Errorf("%v, now falling back to default contract call behaviour that calls latest state", err)
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	latest, finalized, err := m.ht.LatestAndFinalizedBlock(ctx)
```

Comment:
Why do we need to get block numbers for latest and finalized using the `LogPoller` head tracker? The idea behind the confirmations mapping was to allow overrides to the built-in RPC string values such as `latest` and `finalized`. Maybe that wasn't working as expected?

Reply:
We had a discussion, tl;dr we will be piggybacking off of head tracker finality that has recently been exposed as a method.